### PR TITLE
UX-577 Typescript research findings

### PR DESCRIPTION
**Pull request by logansparlin**
### PR Checklist

- [ ] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export * from './components';
```
Comment (jonambas) [inline packages/matchbox/index.d.ts:1]: This doesn't look like a definitions file, should this be `index.ts`?

File: packages/matchbox/package.json [hunk 2]
```diff
@@ -61,7 +64,10 @@
     "rollup-plugin-commonjs": "^10.1.0",
     "rollup-plugin-node-resolve": "^5.2.0",
     "rollup-plugin-terser": "^7.0.2",
-    "styled-components": "^5.2.1"
+    "rollup-plugin-typescript2": "^0.30.0",
+    "styled-components": "^5.2.1",
+    "tslib": "^2.3.0",
+    "typescript": "^4.3.5"
```
Comment (jonambas) [inline packages/matchbox/package.json:70]: Will also need to export definitions from this file `"types": "./types.d.ts"`

**Comment by jonambas** [comment]
wonder if we need to add anything from https://github.com/typescript-eslint/typescript-eslint

File: packages/matchbox/index.d.ts (hunk 1 above)
Comment (logansparlin) [reply packages/matchbox/index.d.ts:1]: Oh right so I just ignored the .d.ts files that were generated just to keep the example PR clean to see the main points. It actually generates a .d.ts file for every component / file and exports to `./components` `./hooks` and `./helpers` so we'd need to wrap that all up in one file or one directory we include so FE has the definitions

File: packages/matchbox/package.json (hunk 2 above)
Comment (logansparlin) [reply packages/matchbox/package.json:70]: Good point

File: packages/matchbox/index.d.ts (hunk 1 above)
Comment (jonambas) [reply packages/matchbox/index.d.ts:1]: Ohhh i see

**Pull request closed without merging**